src/Admin/BlockAdmin.php

```diff
@@ -98,13 +95,12 @@ public function getObject($id)
         return $subject;
     }
 
-    /**
-     * {@inheritdoc}
-     *
-     * @param BaseBlock $object
-     */
     public function preUpdate($object): void
     {
+        if (!$object instanceof DashboardBlockInterface) {
+            throw new \InvalidArgumentException('Invalid block object');
+        }
```

Comment:
Maybe create a dedicated assert* method for this?

Reply:
Did we ever use asserts inside sonat?

Reply:
maybe just a private method `isDashboardBlock`?

Reply:
it is needed in multiple classes, nevermind my suggestion...

Reply:
> Did we ever use asserts inside sonata?

I don't know or think, but you could find another one if you think this isn't really an assertion (i.e. something that runs in dev but not in prod, cf http://php.net/manual/en/function.assert.php

Reply:
Afaik symfony is not using `assert`s anywhere. I think they are not that famous in PHP.

So, can we merge this?